code/krepel/container/array.d

```diff
@@ -209,11 +211,18 @@ struct Array(T)
   void PopFront(size_t Amount = 1)
   {
     DestructArray(Data[0 .. Amount]);
+    if(Count - Amount == 0)
+    {
+      Data = AvailableMemory[0..0];
+    }
+    else
+    {
     Data = Data[Amount .. $];
```

Comment:
Doesn't this work for `Amount == $`? I mean, `AvailableMemory[0 .. 0]` works so I would expect `AvailableMemory[1 .. 1]` to work as well.